pennylane/collections.py

```diff
@@ -290,23 +312,30 @@ def dot(x, y):
     >>> cost(x)
     tensor(-0.2183, dtype=torch.float64, grad_fn=<DotBackward>)
     """
-    if isinstance(x, QNodeCollection) and isinstance(y, QNodeCollection):
+    if hasattr(x, "interface") and hasattr(y, "interface"):
 
         if x.interface != y.interface:
             raise ValueError("QNodeCollections have non-matching interfaces")
 
-        fn = _get_dot_func(x.interface)
-        return lambda params, **kwargs: fn(x(params, **kwargs), y(params, **kwargs))
+        interface = x.interface
```

Comment:
why the extra line?

Reply:
same question applies throughout below

Reply:
Extra line?

Reply:
why define `interface` if it's only used once? why not `some_fn(x.interface)` instead of 
```
interface = x.interface
some_fn(interface)
```
?

Reply:
Oh, because it's used after the if statement for `func.interface = interface`.

Reply:
oh ok, I looked for something like that, but guess I missed it in the github views